packages/cms/pages/resources/resource/resource.tsx

```diff
@@ -192,28 +193,17 @@ export const Table: React.FunctionComponent = () => {
   const [selectedItems, setSelectedItems] = useState<any[]>([])
   const [sortField, setSortField] = useState('firstName')
   const [sortDirection, setSortDirection] = useState<'asc' | 'desc'>('asc')
+  const { resource, applyFilter } = useResourceStore()
 
   const columns: EuiBasicTableColumn<any>[] = [
-    {
-      field: 'firstName',
-      name: 'First name',
-      sortable: true,
-      truncateText: true
-    },
-    {
-      field: 'lastName',
-      name: 'Last name',
-      sortable: true,
-      truncateText: true
-    },
-    {
-      name: 'Date of birth',
-      field: 'dateOfBirth'
-    },
-    {
-      name: 'Nationality',
-      field: 'nationality'
-    },
+    ...(resource?.fields
+      .filter((resource, idx) => resource.showOnIndex && idx != 0)
+      .map(resource => ({
+        field: resource.name,
```

Comment:
So the field here would be the name of how the data is returned from the API, it'll be `field: field.databaseField`.